### ISSUE TITLE
Feat: Add support for Usage API

### ISSUE THREAD

**Issue by SuperPat45** [mudler/LocalAI#4346]
OpenAI added in December 2024, a new Usage API:
https://platform.openai.com/docs/api-reference/usage

> The Usage API provides detailed insights into your activity across the OpenAI API. It also includes a separate [Costs endpoint](https://platform.openai.com/docs/api-reference/usage/costs), which offers visibility into your spend, breaking down consumption by invoice line items and project IDs.

It would be interesting to have it in LocalAI

**Comment by jtwolfe** [comment]
good idea but imo would be included in the list of 'enterprise-like' features that include, model-to-user/api controls that I would like to see.
